x/concentrated-liquidity/internal/math/tick_test.go

```diff
@@ -6,40 +6,101 @@ import (
-			tickIndex:         sdk.NewInt(-86129),
-			sqrtPriceExpected: "0.013484226394978088",
-			// https://www.wolframalpha.com/input?i2d=true&i=Power%5B1.0001%2CDivide%5B-86129%2C2%5D%5D
+		"One dollar increments at the ten thousands place: 1": {
+			tickIndex:     sdk.NewInt(400000),
+			kAtPriceOne:   sdk.NewInt(-4),
+			expectedPrice: "50000.000000000000000000",
+		},
+		"One dollar increments at the ten thousands place: 2": {
+			tickIndex:     sdk.NewInt(400001),
+			kAtPriceOne:   sdk.NewInt(-4),
+			expectedPrice: "50001.000000000000000000",
+		},
+		"Ten cent increments at the ten thousands place: 1": {
+			tickIndex:     sdk.NewInt(4000000),
+			kAtPriceOne:   sdk.NewInt(-5),
+			expectedPrice: "50000.000000000000000000",
+		},
+		"Ten cent increments at the ten thousands place: 2": {
+			tickIndex:     sdk.NewInt(4000001),
+			kAtPriceOne:   sdk.NewInt(-5),
+			expectedPrice: "50000.100000000000000000",
+		},
+		"One cent increments at the ten thousands place: 1": {
+			tickIndex:     sdk.NewInt(40000000),
+			kAtPriceOne:   sdk.NewInt(-6),
+			expectedPrice: "50000.000000000000000000",
+		},
+		"One cent increments at the ten thousands place: 2": {
+			tickIndex:     sdk.NewInt(40000001),
+			kAtPriceOne:   sdk.NewInt(-6),
+			expectedPrice: "50000.010000000000000000",
+		},
+		"One cent increments at the ones place: 1": {
+			tickIndex:     sdk.NewInt(400),
+			kAtPriceOne:   sdk.NewInt(-2),
+			expectedPrice: "5.000000000000000000",
+		},
+		"One cent increments at the ones place: 2": {
+			tickIndex:     sdk.NewInt(401),
+			kAtPriceOne:   sdk.NewInt(-2),
+			expectedPrice: "5.010000000000000000",
+		},
+		"Ten cent increments at the tens place: 1": {
+			tickIndex:     sdk.NewInt(1300),
+			kAtPriceOne:   sdk.NewInt(-2),
+			expectedPrice: "50.000000000000000000",
+		},
+		"Ten cent increments at the ones place: 2": {
+			tickIndex:     sdk.NewInt(1301),
+			kAtPriceOne:   sdk.NewInt(-2),
+			expectedPrice: "50.100000000000000000",
+		},
+		"One cent increments at the tenths place: 1": {
+			tickIndex:     sdk.NewInt(-2),
+			kAtPriceOne:   sdk.NewInt(-2),
+			expectedPrice: "0.980000000000000000",
+		},
+		"One tenth of a cent increments at the hundredths place: 1": {
+			tickIndex:     sdk.NewInt(-2),
+			kAtPriceOne:   sdk.NewInt(-3),
+			expectedPrice: "0.998000000000000000",
+		},
+		"One hundredths of a cent increments at the thousandths place: 1": {
+			tickIndex:     sdk.NewInt(-2),
+			kAtPriceOne:   sdk.NewInt(-4),
+			expectedPrice: "0.999800000000000000",
+		},
+		"One ten millionth of a cent increments at the hundred millionths place: 1": {
+			tickIndex:     sdk.NewInt(-2),
+			kAtPriceOne:   sdk.NewInt(-8),
+			expectedPrice: "0.999999980000000000",
+		},
+		"One ten millionth of a cent increments at the hundred millionths place: 2": {
+			tickIndex:     sdk.NewInt(-99999111),
+			kAtPriceOne:   sdk.NewInt(-8),
+			expectedPrice: "0.000008890000000000",
+		},
+		"More variety of numbers in each place": {
```

Comment:
Can we add tests at each boundary please

Reply:
By each boundary, do you mean select the minimum k value possible (in this case -12) and add a test case for each increment until we reach the maximum k value?